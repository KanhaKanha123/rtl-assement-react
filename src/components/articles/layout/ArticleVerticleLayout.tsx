import { Card } from 'react-bootstrap';
import { BundlesProps } from "../../../types/types";

type ArticleVerticleLayoutProps = {
    bundle: BundlesProps
}

const ArticleVerticleLayout = ({ bundle: { afbeelding, urlAlias, labelValue, title, lead, formatted } }: ArticleVerticleLayoutProps) => {
    return (
        <div data-testid="article-verticle-layout-wrapper" className="article-verticle-layout">
            <Card>
                <a data-testid="verticleLayout-a" href={urlAlias} target="_blank">
                    <Card.Img data-testid="verticleLayout-img" variant='top' style={{ height: '250px' }} src={afbeelding} alt={labelValue}></Card.Img>
                </a>
                <Card.Title data-testid="verticleLayout-label-title" className="label">{labelValue}</Card.Title>
                <Card.Title data-testid="verticleLayout-label-date" className="label">{formatted}</Card.Title>
                <span data-testid="verticleLayout-layoutBoldTitle" className='layout-bold-title'>{title}</span>
                <span data-testid="verticleLayout-lead" className="lead">{lead.substring(0, 30)}...</span>
            </Card>
        </div >

    )
}

export default ArticleVerticleLayout;