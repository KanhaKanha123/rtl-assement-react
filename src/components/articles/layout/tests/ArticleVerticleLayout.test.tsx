import ArticleVerticleLayout from '../ArticleVerticleLayout';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import * as testData from './test.data.json';

describe('ArticleVerticleLayout component test', () => {

    test("ArticleVerticleLayout component load correctly", () => {
        render(<ArticleVerticleLayout bundle={testData.withData}/>);

        const anchorTag=screen.getByTestId("verticleLayout-a");
        const wrapper=screen.getByTestId("article-verticle-layout-wrapper");
        const verticleLayoutImage=screen.getByTestId("verticleLayout-img");
        const verticleLayoutLabelTitle=screen.getByTestId("verticleLayout-label-title");
        const verticleLayoutLabelDate=screen.getByTestId("verticleLayout-label-date");
        const verticleLayoutBoldTitle=screen.getByTestId("verticleLayout-layoutBoldTitle");
        const verticleLayoutLead=screen.getByTestId("verticleLayout-lead");
       
       //check element exist
        expect(wrapper).toBeInTheDocument();
        expect(anchorTag).toBeInTheDocument();
        expect(verticleLayoutImage).toBeInTheDocument();
        expect(verticleLayoutLabelTitle).toBeInTheDocument();
        expect(verticleLayoutLabelDate).toBeInTheDocument();
        expect(verticleLayoutBoldTitle).toBeInTheDocument();
        expect(verticleLayoutLead).toBeInTheDocument();
    });

    test("ArticleVerticleLayout component recieve correct props", () => {
        render(<ArticleVerticleLayout bundle={testData.withData}/>);

        const verticleLayoutLabelTitle=screen.getByTestId("verticleLayout-label-title");
        const verticleLayoutLabelDate=screen.getByTestId("verticleLayout-label-date");
        const verticleLayoutBoldTitle=screen.getByTestId("verticleLayout-layoutBoldTitle");
        const verticleLayoutLead=screen.getByTestId("verticleLayout-lead");
        
        //match values
        expect(verticleLayoutLabelTitle).toHaveTextContent('De uitschieters van 2018');
        expect(verticleLayoutLabelDate).toHaveTextContent('19 juni 2018');
        expect(verticleLayoutBoldTitle).toHaveTextContent('Trijntje');
        expect(verticleLayoutLead).toHaveTextContent('Humberto Tan');
    });

    test("ArticleVerticleLayout component not recieve correct props", () => {
        render(<ArticleVerticleLayout bundle={testData.withoutData}/>);
        
        const verticleLayoutLabelTitle=screen.getByTestId("verticleLayout-label-title");
        const verticleLayoutLabelDate=screen.getByTestId("verticleLayout-label-date");
        const verticleLayoutBoldTitle=screen.getByTestId("verticleLayout-layoutBoldTitle");
        const verticleLayoutLead=screen.getByTestId("verticleLayout-lead");
        
        //match values
        expect(verticleLayoutLabelTitle).toBeEmptyDOMElement();
        expect(verticleLayoutLabelDate).toBeEmptyDOMElement();
        expect(verticleLayoutBoldTitle).toBeEmptyDOMElement();
        expect(verticleLayoutLead).toHaveTextContent('...');
    });

    test("ArticleVerticleLayout should match the snapshot", () => {
        const element=renderer.create(<ArticleVerticleLayout bundle={testData.withData}/>);
        expect(element.toJSON()).toMatchSnapshot();
});
})
