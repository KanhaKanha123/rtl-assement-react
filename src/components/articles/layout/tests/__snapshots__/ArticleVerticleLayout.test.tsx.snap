// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleVerticleLayout component test ArticleVerticleLayout should match the snapshot 1`] = `
<div
  className="article-verticle-layout"
  data-testid="article-verticle-layout-wrapper"
>
  <div
    className="card"
  >
    <a
      data-testid="verticleLayout-a"
      href="https://www.rtlboulevard.nl/entertainment/showbizz/artikel/4245471/trijntje-oosterhuis-onherkenbaar-veel-afgevallen"
      target="_blank"
    >
      <img
        alt="De uitschieters van 2018"
        className="card-img-top"
        data-testid="verticleLayout-img"
        src="https://www.rtlboulevard.nl/sites/default/files/content/images/2018/06/19/ANP-56919532.jpg"
        style={
          Object {
            "height": "250px",
          }
        }
      />
    </a>
    <div
      className="label card-title h5"
      data-testid="verticleLayout-label-title"
    >
      De uitschieters van 2018
    </div>
    <div
      className="label card-title h5"
      data-testid="verticleLayout-label-date"
    >
      19 juni 2018
    </div>
    <span
      className="layout-bold-title"
      data-testid="verticleLayout-layoutBoldTitle"
    >
      Trijntje
    </span>
    <span
      className="lead"
      data-testid="verticleLayout-lead"
    >
      Humberto Tan
      ...
    </span>
  </div>
</div>
`;
